httptools/parser/url_parser.pyx

```diff
@@ -0,0 +1,108 @@
+#cython: language_level=3
+
+from __future__ import print_function
+from cpython.mem cimport PyMem_Malloc, PyMem_Free
+from cpython cimport PyObject_GetBuffer, PyBuffer_Release, PyBUF_SIMPLE, \
+                     Py_buffer
+
+from .errors import HttpParserInvalidURLError
+
+cimport cython
+from . cimport uparser
```

Comment:
The name `uparser` feels weird to me comparing to `cparser.pxd` standing for "the Cython header file to the parser written in C". Perhaps `url_cparser` is more aligned, if we don't want to name the pxd files after llhttp or http-parser for good reasons like using future llurl for `url_cparser`. So let's have:
 * `parser.pyx` -> `cparser.pxd` for HTTP parsing
 * `url_parser.pyx` -> `url_cparser.pxd` for URL parsing